wzk/net/XhrFactory.coffee

```diff
@@ -60,7 +75,37 @@ class wzk.net.XhrFactory
       @flash.clearAll() if config.flash
       @applyJsonResponse response, config, xhr.getStatus()
     else
-      @flash.error(xhr.getStatus()) if e.target.getConfig().flash
+      @handleCriticalError xhr, e.target.getConfig().flash
+
+
+  ###*
+    @protected
+    @param {wzk.net.XhrIo} xhr
+    @param {wzk.net.FlashMiddleware} flash
+  ###
+  handleCriticalError: (xhr, flash) ->
+    return if not flash
+    isOnline = @isOnline()
+    if xhr.getStatus() is 0 and isOnline is false and @wasOnline()
+      @flash.offline()
+    else if xhr.getStatus() is 500
+      @flash.error xhr.getStatus()
```

Comment:
Messages translation for messages must be added to base template.

Offline by default says Offline, flash.error says Interní chyba, prosím, opakujte později.